apps/server/src/modules/tool/uc/external-tool.uc.ts

```diff
@@ -1,77 +1,25 @@
-import { Inject, Injectable, UnprocessableEntityException } from '@nestjs/common';
-import { IFindOptions, Permission, User } from '@shared/domain';
-import { ExternalToolDO, Lti11ToolConfigDO, Oauth2ToolConfigDO } from '@shared/domain/domainobject/external-tool';
+import { Injectable, UnprocessableEntityException } from '@nestjs/common';
+import { EntityId, IFindOptions, Permission, User } from '@shared/domain';
+import { ExternalToolDO, Oauth2ToolConfigDO } from '@shared/domain/domainobject/external-tool';
 import { AuthorizationService } from '@src/modules/authorization';
-import { OauthProviderService } from '@shared/infra/oauth-provider';
-import { ProviderOauthClient } from '@shared/infra/oauth-provider/dto';
-import { DefaultEncryptionService, IEncryptionService } from '@shared/infra/encryption';
 import { Page } from '@shared/domain/interface/page';
 import { ExternalToolService } from '../service/external-tool.service';
-import { ExternalToolRequestMapper } from '../mapper/external-tool-request.mapper';
 
 @Injectable()
 export class ExternalToolUc {
 	constructor(
 		private readonly externalToolService: ExternalToolService,
-		private readonly externalToolMapper: ExternalToolRequestMapper,
-		private readonly authorizationService: AuthorizationService,
-		private readonly oauthProviderService: OauthProviderService,
-		@Inject(DefaultEncryptionService) private readonly oAuthEncryptionService: IEncryptionService
+		private readonly authorizationService: AuthorizationService
 	) {}
 
-	async createExternalTool(externalToolDO: ExternalToolDO, userId: string): Promise<ExternalToolDO> {
+	async createExternalTool(userId: EntityId, externalToolDO: ExternalToolDO): Promise<ExternalToolDO> {
```

Comment:
We should not use domain objects as use case parameters. That has been discussed in the server arch group.
https://docs.dbildungscloud.de/display/DBH/ARC+Meeting+2022-12-01

Reply:
Thanks for the hint. In future stories/estimates/PRs we will take that, but in my opinion we can't implement such big architectural decisions during the sprint.

Reply:
Okay, we can move that to the next sprint.